### ISSUE TITLE
app.js

### ISSUE THREAD

**Issue by altintx** [Journerist/gulp-extify#1]
Do you have a vision for how app.js should be handled? (app.js being special because it doesn't define a class.) Feel like the lowest work answer is to put the contents of app.js in my wrapper HTML file, but also considered emitting a ```Ext.create("Namespace.Application", {});``` and the end of afterFileCollection. Not sure right direction to go.

**Comment by Journerist** [comment]
It's up to you to decide whats the best way to start your application. At least you have to instantiate your application like this:
```JavaScript
Ext.onReady(function() {
	Ext.getBody().mask();

	// No animations
	Ext.enableFx = false;

	Ext.application({
		name: 'AppName',
		controllers: [
			'Namespace.controller.Root'
		],
		launch: function() {
			//do whatever you want
		}
	});
	Ext.getBody().unmask();
});
```

You can do it in your index.html or in an included file. There is no need to add your app.js to your gulp built process at all.

If you have no other questions about handling the generated app.js then I will close this ticket within 7 days.

**Comment by altintx** [comment]
Did the embed in .html to bootstrap app, and it worked.

**Comment by stevemwasserman** [comment]
Can you explain the work around with app.js and Application.js when structuring your app just like it can be generated with Sencha Cmd?  I am able to concatenate my app, but I am receiving 404 errors, which I believe to be related to the ordering of the files.  I understand that this package will eliminate this issue.

**Comment by Journerist** [comment]
The purpose of this tool is simple: concatenate the a given file and it's dependencies correctly. Also, this is a small subset of sechacmd's features.

Most of the time errors like 404 are occuring because of missing dependency definitions. Sencha does some black magic and I guess you had luck that your application worked. Back in the days I did the following to trace the issue down:

1. Build the files using this tool
2. Check what file is missing
3. Search the file in the bundled file. You will realize that it's way to late in the file or even not included. Figure out what files require the missing file and add the missing dependency.
4. Build again and go to step 2 if necessary

Basically, it's only JavaScript. You framework needs to be loaded, you application needs to be available. Afterwards, you need to trigger somehow ExtJs to start the application. That's it.

Good luck :)
